{"matches":[{"vulnerability":{"id":"GHSA-pq64-v7f5-gqh8","dataSource":"https://github.com/advisories/GHSA-pq64-v7f5-gqh8","namespace":"github:language:python","severity":"High","urls":[],"description":"Pygments vulnerable to Regular Expression Denial of Service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-27291","epss":0.0252,"percentile":0.847,"date":"2025-06-03"}],"fix":{"versions":["2.7.4"],"state":"fixed"},"advisories":[],"risk":1.8900000000000001},"relatedVulnerabilities":[{"id":"CVE-2021-27291","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-27291","namespace":"nvd:cpe","severity":"High","urls":["https://gist.github.com/b-c-ds/b1a2cc0c68a35c57188575eb496de5ce","https://github.com/pygments/pygments/commit/2e7e8c4a7b318f4032493773732754e418279a14","https://lists.debian.org/debian-lts-announce/2021/03/msg00024.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00003.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00006.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/GSJRFHALQ7E3UV4FFMFU2YQ6LUDHAI55/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/WSLD67LFGXOX2K5YNESSWAS4AGZIJTUQ/","https://www.debian.org/security/2021/dsa-4878","https://www.debian.org/security/2021/dsa-4889","https://gist.github.com/b-c-ds/b1a2cc0c68a35c57188575eb496de5ce","https://github.com/pygments/pygments/commit/2e7e8c4a7b318f4032493773732754e418279a14","https://lists.debian.org/debian-lts-announce/2021/03/msg00024.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00003.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00006.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/GSJRFHALQ7E3UV4FFMFU2YQ6LUDHAI55/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/WSLD67LFGXOX2K5YNESSWAS4AGZIJTUQ/","https://www.debian.org/security/2021/dsa-4878","https://www.debian.org/security/2021/dsa-4889"],"description":"In pygments 1.1+, fixed in 2.7.4, the lexers used to parse programming languages rely heavily on regular expressions. Some of the regular expressions have exponential or cubic worst-case complexity and are vulnerable to ReDoS. By crafting malicious input, an attacker can cause a denial of service.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-27291","epss":0.0252,"percentile":0.847,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pygments","version":"2.2.0"}},"found":{"versionConstraint":">=1.1,<2.7.4 (python)","vulnerabilityID":"GHSA-pq64-v7f5-gqh8"},"fix":{"suggestedVersion":"2.7.4"}}],"artifact":{"id":"283eb5d8c91da60d","name":"pygments","version":"2.2.0","type":"python","locations":[{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD License"],"cpes":["cpe:2.3:a:georg_brandl_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:pygments:2.2.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pygments@2.2.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-pq7m-3gw7-gq5x","dataSource":"https://github.com/advisories/GHSA-pq7m-3gw7-gq5x","namespace":"github:language:python","severity":"High","urls":[],"description":"Execution with Unnecessary Privileges in ipython","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:R/S:C/C:H/I:H/A:H","metrics":{"baseScore":8.2,"exploitabilityScore":1.5,"impactScore":6.1},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-21699","epss":0.01458,"percentile":0.79929,"date":"2025-06-03"}],"fix":{"versions":["5.11"],"state":"fixed"},"advisories":[],"risk":1.1445299999999998},"relatedVulnerabilities":[{"id":"CVE-2022-21699","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-21699","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/ipython/ipython/commit/46a51ed69cdf41b4333943d9ceeb945c4ede5668","https://github.com/ipython/ipython/security/advisories/GHSA-pq7m-3gw7-gq5x","https://ipython.readthedocs.io/en/stable/whatsnew/version8.html#ipython-8-0-1-cve-2022-21699","https://lists.debian.org/debian-lts-announce/2022/01/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CRQRTWHYXMLDJ572VGVUZMUPEOTPM3KB/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/DZ7LVZBB4D7KVSFNEQUBEHFO3JW6D2ZK/","https://github.com/ipython/ipython/commit/46a51ed69cdf41b4333943d9ceeb945c4ede5668","https://github.com/ipython/ipython/security/advisories/GHSA-pq7m-3gw7-gq5x","https://ipython.readthedocs.io/en/stable/whatsnew/version8.html#ipython-8-0-1-cve-2022-21699","https://lists.debian.org/debian-lts-announce/2022/01/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CRQRTWHYXMLDJ572VGVUZMUPEOTPM3KB/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/DZ7LVZBB4D7KVSFNEQUBEHFO3JW6D2ZK/"],"description":"IPython (Interactive Python) is a command shell for interactive computing in multiple programming languages, originally developed for the Python programming language. Affected versions are subject to an arbitrary code execution vulnerability achieved by not properly managing cross user temporary files. This vulnerability allows one user to run code as another on the same machine. All users are advised to upgrade.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:C/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.1,"impactScore":6.1},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:L/AC:L/Au:N/C:P/I:P/A:P","metrics":{"baseScore":4.6,"exploitabilityScore":4,"impactScore":6.5},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:R/S:C/C:H/I:H/A:H","metrics":{"baseScore":8.2,"exploitabilityScore":1.5,"impactScore":6.1},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-21699","epss":0.01458,"percentile":0.79929,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"ipython","version":"5.5.0"}},"found":{"versionConstraint":"<5.11 (python)","vulnerabilityID":"GHSA-pq7m-3gw7-gq5x"},"fix":{"suggestedVersion":"5.11"}}],"artifact":{"id":"1d150410f0d89c5e","name":"ipython","version":"5.5.0","type":"python","locations":[{"path":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/top_level.txt","accessPath":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:ipython_development_team_project:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team_project:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team_project:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:ipython:5.5.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/ipython@5.5.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-gpvv-69j7-gwj8","dataSource":"https://github.com/advisories/GHSA-gpvv-69j7-gwj8","namespace":"github:language:python","severity":"High","urls":[],"description":"Path Traversal in pip","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2019-20916","epss":0.00622,"percentile":0.69107,"date":"2025-06-03"}],"fix":{"versions":["19.2"],"state":"fixed"},"advisories":[],"risk":0.46649999999999997},"relatedVulnerabilities":[{"id":"CVE-2019-20916","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2019-20916","namespace":"nvd:cpe","severity":"High","urls":["http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00005.html","http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00010.html","https://github.com/gzpan123/pip/commit/a4c735b14a62f9cb864533808ac63936704f2ace","https://github.com/pypa/pip/compare/19.1.1...19.2","https://github.com/pypa/pip/issues/6413","https://lists.debian.org/debian-lts-announce/2020/09/msg00010.html","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html","http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00005.html","http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00010.html","https://github.com/gzpan123/pip/commit/a4c735b14a62f9cb864533808ac63936704f2ace","https://github.com/pypa/pip/compare/19.1.1...19.2","https://github.com/pypa/pip/issues/6413","https://lists.debian.org/debian-lts-announce/2020/09/msg00010.html","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html"],"description":"The pip package before 19.2 for Python allows Directory Traversal when a URL is given in an install command, because a Content-Disposition header can have ../ in a filename, as demonstrated by overwriting the /root/.ssh/authorized_keys file. This occurs in _download_http_url in _internal/download.py.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:P/A:N","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2019-20916","epss":0.00622,"percentile":0.69107,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"19.1.1"}},"found":{"versionConstraint":"<19.2 (python)","vulnerabilityID":"GHSA-gpvv-69j7-gwj8"},"fix":{"suggestedVersion":"19.2"}}],"artifact":{"id":"89caa1edd580f6b3","name":"pip","version":"19.1.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:19.1.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@19.1.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"41.0.1"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"8f69968e2ab1fd24","name":"setuptools","version":"41.0.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:41.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@41.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-v3c5-jqr6-7qm8","dataSource":"https://github.com/advisories/GHSA-v3c5-jqr6-7qm8","namespace":"github:language:python","severity":"High","urls":[],"description":"Python Charmers Future denial of service vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40899","epss":0.00419,"percentile":0.61102,"date":"2025-06-03"}],"fix":{"versions":["0.18.3"],"state":"fixed"},"advisories":[],"risk":0.31425000000000003},"relatedVulnerabilities":[{"id":"CVE-2022-40899","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40899","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215","https://github.com/PythonCharmers/python-future/pull/610","https://github.com/python/cpython/pull/17157","https://pypi.org/project/future/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215","https://github.com/PythonCharmers/python-future/pull/610","https://github.com/python/cpython/pull/17157","https://pypi.org/project/future/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"],"description":"An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40899","epss":0.00419,"percentile":0.61102,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"future","version":"0.17.1"}},"found":{"versionConstraint":"<=0.18.2 (python)","vulnerabilityID":"GHSA-v3c5-jqr6-7qm8"},"fix":{"suggestedVersion":"0.18.3"}}],"artifact":{"id":"f7899a9983f66a38","name":"future","version":"0.17.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/future-0.17.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/future-0.17.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/future-0.17.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/future-0.17.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:ed_schofield_project:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield_project:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield_project:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python-future:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python-future:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python_future:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python_future:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:edproject:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:edproject:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:future:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:future:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python-future:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python_future:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed:python-future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed:python_future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:edproject:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:future:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:python:future:0.17.1:*:*:*:*:*:*:*","cpe:2.3:a:ed:future:0.17.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/future@0.17.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-r9hx-vwmv-q579","dataSource":"https://github.com/advisories/GHSA-r9hx-vwmv-q579","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54328,"date":"2025-06-03"}],"fix":{"versions":["65.5.1"],"state":"fixed"},"advisories":[],"risk":0.2385},"relatedVulnerabilities":[{"id":"CVE-2022-40897","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40897","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/","https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/"],"description":"Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54328,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"41.0.1"}},"found":{"versionConstraint":"<65.5.1 (python)","vulnerabilityID":"GHSA-r9hx-vwmv-q579"},"fix":{"suggestedVersion":"65.5.1"}}],"artifact":{"id":"8f69968e2ab1fd24","name":"setuptools","version":"41.0.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:41.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@41.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"41.0.1"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"8f69968e2ab1fd24","name":"setuptools","version":"41.0.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/setuptools-41.0.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:41.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@41.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-5xp3-jfq3-5q8x","dataSource":"https://github.com/advisories/GHSA-5xp3-jfq3-5q8x","namespace":"github:language:python","severity":"High","urls":[],"description":"Improper Input Validation in pip","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-3572","epss":0.0024,"percentile":0.4753,"date":"2025-06-03"}],"fix":{"versions":["21.1"],"state":"fixed"},"advisories":[],"risk":0.15839999999999999},"relatedVulnerabilities":[{"id":"CVE-2021-3572","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-3572","namespace":"nvd:cpe","severity":"Medium","urls":["https://bugzilla.redhat.com/show_bug.cgi?id=1962856","https://security.netapp.com/advisory/ntap-20240621-0006/","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html","https://bugzilla.redhat.com/show_bug.cgi?id=1962856","https://security.netapp.com/advisory/ntap-20240621-0006/","https://www.oracle.com/security-alerts/cpuapr2022.html","https://www.oracle.com/security-alerts/cpujul2022.html"],"description":"A flaw was found in python-pip in the way it handled Unicode separators in git references. A remote attacker could possibly use this issue to install a different revision on a repository. The highest threat from this vulnerability is to data integrity. This is fixed in python-pip version 21.1.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:M/Au:S/C:N/I:P/A:N","metrics":{"baseScore":3.5,"exploitabilityScore":6.9,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-3572","epss":0.0024,"percentile":0.4753,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"19.1.1"}},"found":{"versionConstraint":"<21.1 (python)","vulnerabilityID":"GHSA-5xp3-jfq3-5q8x"},"fix":{"suggestedVersion":"21.1"}}],"artifact":{"id":"89caa1edd580f6b3","name":"pip","version":"19.1.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:19.1.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@19.1.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-29gw-9793-fvw7","dataSource":"https://github.com/advisories/GHSA-29gw-9793-fvw7","namespace":"github:language:python","severity":"Low","urls":[],"description":"IPython vulnerable to command injection via set_term_title","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:L","metrics":{"baseScore":4.5,"exploitabilityScore":1.1,"impactScore":3.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-24816","epss":0.00417,"percentile":0.60949,"date":"2025-06-03"}],"fix":{"versions":["8.10.0"],"state":"fixed"},"advisories":[],"risk":0.15637500000000001},"relatedVulnerabilities":[{"id":"CVE-2023-24816","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-24816","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117","https://github.com/ipython/ipython/blob/56e6925dfa50e2c7f4a6471547b8176275db7c25/IPython/utils/_process_win32.py#L20","https://github.com/ipython/ipython/commit/385d69325319a5972ee9b5983638e3617f21cb1f","https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7","https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117","https://github.com/ipython/ipython/blob/56e6925dfa50e2c7f4a6471547b8176275db7c25/IPython/utils/_process_win32.py#L20","https://github.com/ipython/ipython/commit/385d69325319a5972ee9b5983638e3617f21cb1f","https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7"],"description":"IPython (Interactive Python) is a command shell for interactive computing in multiple programming languages, originally developed for the Python programming language. Versions prior to 8.1.0 are subject to a command injection vulnerability with very specific prerequisites. This vulnerability requires that the function `IPython.utils.terminal.set_term_title` be called on Windows in a Python environment where ctypes is not available. The dependency on `ctypes` in `IPython.utils._process_win32` prevents the vulnerable code from ever being reached in the ipython binary. However, as a library that could be used by another tool `set_term_title` could be called and hence introduce a vulnerability. Should an attacker get untrusted input to an instance of this function they would be able to inject shell commands as current process and limited to the scope of the current process. Users of ipython as a library are advised to upgrade. Users unable to upgrade should ensure that any calls to the `IPython.utils.terminal.set_term_title` function are done with trusted or filtered input.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7,"exploitabilityScore":1.1,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:L","metrics":{"baseScore":4.5,"exploitabilityScore":1.1,"impactScore":3.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-24816","epss":0.00417,"percentile":0.60949,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"ipython","version":"5.5.0"}},"found":{"versionConstraint":"<8.10.0 (python)","vulnerabilityID":"GHSA-29gw-9793-fvw7"},"fix":{"suggestedVersion":"8.10.0"}}],"artifact":{"id":"1d150410f0d89c5e","name":"ipython","version":"5.5.0","type":"python","locations":[{"path":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/top_level.txt","accessPath":"/usr/lib/python2.7/dist-packages/ipython-5.5.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:ipython_development_team_project:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team_project:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team_project:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_teamproject:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_development_team:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev_project:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_devproject:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python-ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python_ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython-dev:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython_dev:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:ipython:ipython:5.5.0:*:*:*:*:*:*:*","cpe:2.3:a:python:ipython:5.5.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/ipython@5.5.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-qwmp-2cf2-g9g6","dataSource":"https://github.com/advisories/GHSA-qwmp-2cf2-g9g6","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/wheel vulnerable to Regular Expression denial of service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30278,"date":"2025-06-03"}],"fix":{"versions":["0.38.1"],"state":"fixed"},"advisories":[],"risk":0.08099999999999999},"relatedVulnerabilities":[{"id":"CVE-2022-40898","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40898","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"],"description":"An issue discovered in Python Packaging Authority (PyPA) Wheel 0.37.1 and earlier allows remote attackers to cause a denial of service via attacker controlled input to wheel cli.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30278,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"wheel","version":"0.33.4"}},"found":{"versionConstraint":"<0.38.1 (python)","vulnerabilityID":"GHSA-qwmp-2cf2-g9g6"},"fix":{"suggestedVersion":"0.38.1"}}],"artifact":{"id":"00f01c21eeb1cd42","name":"wheel","version":"0.33.4","type":"python","locations":[{"path":"/lib/python2.7/site-packages/wheel-0.33.4.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/wheel-0.33.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/wheel-0.33.4.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/wheel-0.33.4.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:daniel_holth_project:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python-wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python_wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:dholth:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:python:wheel:0.33.4:*:*:*:*:*:*:*","cpe:2.3:a:wheel:wheel:0.33.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/wheel@0.33.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-9w8r-397f-prfh","dataSource":"https://github.com/advisories/GHSA-9w8r-397f-prfh","namespace":"github:language:python","severity":"High","urls":[],"description":"Infinite Loop in Pygments","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-20270","epss":0.00097,"percentile":0.2828,"date":"2025-06-03"}],"fix":{"versions":["2.7.4"],"state":"fixed"},"advisories":[],"risk":0.07275000000000001},"relatedVulnerabilities":[{"id":"CVE-2021-20270","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-20270","namespace":"nvd:cpe","severity":"High","urls":["https://bugzilla.redhat.com/show_bug.cgi?id=1922136","https://lists.debian.org/debian-lts-announce/2021/05/msg00003.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00006.html","https://www.debian.org/security/2021/dsa-4889","https://www.oracle.com/security-alerts/cpuoct2021.html","https://bugzilla.redhat.com/show_bug.cgi?id=1922136","https://lists.debian.org/debian-lts-announce/2021/05/msg00003.html","https://lists.debian.org/debian-lts-announce/2021/05/msg00006.html","https://www.debian.org/security/2021/dsa-4889","https://www.oracle.com/security-alerts/cpuoct2021.html"],"description":"An infinite loop in SMLLexer in Pygments versions 1.5 to 2.7.3 may lead to denial of service when performing syntax highlighting of a Standard ML (SML) source file, as demonstrated by input that only contains the \"exception\" keyword.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-20270","epss":0.00097,"percentile":0.2828,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pygments","version":"2.2.0"}},"found":{"versionConstraint":">=1.5,<2.7.4 (python)","vulnerabilityID":"GHSA-9w8r-397f-prfh"},"fix":{"suggestedVersion":"2.7.4"}}],"artifact":{"id":"283eb5d8c91da60d","name":"pygments","version":"2.2.0","type":"python","locations":[{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD License"],"cpes":["cpe:2.3:a:georg_brandl_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:pygments:2.2.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pygments@2.2.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-mq26-g339-26xf","dataSource":"https://github.com/advisories/GHSA-mq26-g339-26xf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Command Injection in pip when used with Mercurial","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11691,"date":"2025-06-03"}],"fix":{"versions":["23.3"],"state":"fixed"},"advisories":[],"risk":0.021},"relatedVulnerabilities":[{"id":"CVE-2023-5752","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-5752","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/","https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"],"description":"When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11691,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"19.1.1"}},"found":{"versionConstraint":"<23.3 (python)","vulnerabilityID":"GHSA-mq26-g339-26xf"},"fix":{"suggestedVersion":"23.3"}}],"artifact":{"id":"89caa1edd580f6b3","name":"pip","version":"19.1.1","type":"python","locations":[{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","accessPath":"/lib/python2.7/site-packages/pip-19.1.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev_project:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_devproject:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa-dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa_dev:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:19.1.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:19.1.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@19.1.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-mrwq-x4v8-fh7p","dataSource":"https://github.com/advisories/GHSA-mrwq-x4v8-fh7p","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Pygments vulnerable to ReDoS","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40896","epss":0.00034,"percentile":0.0835,"date":"2025-06-03"}],"fix":{"versions":["2.15.0"],"state":"fixed"},"advisories":[],"risk":0.01785},"relatedVulnerabilities":[{"id":"CVE-2022-40896","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40896","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pygments/pygments/blob/master/pygments/lexers/smithy.py#L61","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/EZGMXALE3HSP4OXC7UUWIKX3OXKZDTY3/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VUZO4BQCIY2S2KZYHERQMKURB7AHXDBO/","https://pypi.org/project/Pygments/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages-part-2/","https://github.com/pygments/pygments/blob/master/pygments/lexers/smithy.py#L61","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/EZGMXALE3HSP4OXC7UUWIKX3OXKZDTY3/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VUZO4BQCIY2S2KZYHERQMKURB7AHXDBO/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/EZGMXALE3HSP4OXC7UUWIKX3OXKZDTY3/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VUZO4BQCIY2S2KZYHERQMKURB7AHXDBO/","https://pypi.org/project/Pygments/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages-part-2/"],"description":"A ReDoS issue was discovered in pygments/lexers/smithy.py in pygments through 2.15.0 via SmithyLexer.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40896","epss":0.00034,"percentile":0.0835,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pygments","version":"2.2.0"}},"found":{"versionConstraint":"<2.15.0 (python)","vulnerabilityID":"GHSA-mrwq-x4v8-fh7p"},"fix":{"suggestedVersion":"2.15.0"}}],"artifact":{"id":"283eb5d8c91da60d","name":"pygments","version":"2.2.0","type":"python","locations":[{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","accessPath":"/usr/lib/python2.7/dist-packages/Pygments-2.2.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD License"],"cpes":["cpe:2.3:a:georg_brandl_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandlproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_project:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python-pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:python_pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg_brandl:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georgproject:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pygments:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pygments:2.2.0:*:*:*:*:*:*:*","cpe:2.3:a:georg:pygments:2.2.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pygments@2.2.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"rfcat-phocean-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/rfcat-phocean-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/rfcat-phocean-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-04T01:32:23Z_1749035557.tar.zst?checksum=sha256%3A77238f4ffbc7cf72bf4dedabbb21a3d0dd3ade90fb5d9df7c4da2870f8d1a9b7","built":"2025-06-04T11:12:37Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-04T01:32:23Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-04T01:32:32Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-04T01:32:33Z","input":"xxh64:03c9c4d182d77791"},"debian":{"captured":"2025-06-04T01:32:49Z","input":"xxh64:169ceb8fbcb8d6f1"},"epss":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:bf87c4c234d02346"},"github":{"captured":"2025-06-04T01:32:30Z","input":"xxh64:3dab0be50635e057"},"kev":{"captured":"2025-06-04T01:32:45Z","input":"xxh64:a934e75f468f4e09"},"mariner":{"captured":"2025-06-04T01:32:43Z","input":"xxh64:ac698752085c186f"},"nvd":{"captured":"2025-06-04T01:33:28Z","input":"xxh64:389887843a59d29c"},"oracle":{"captured":"2025-06-04T01:32:35Z","input":"xxh64:0b8aabcfb6505b62"},"rhel":{"captured":"2025-06-04T04:49:10Z","input":"xxh64:947d0caeb19e9ccc"},"sles":{"captured":"2025-06-04T01:32:34Z","input":"xxh64:1c50b1219296628a"},"ubuntu":{"captured":"2025-06-04T01:33:14Z","input":"xxh64:71e22d5c5bbe7d14"},"wolfi":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:e68fca5a870c6769"}}},"timestamp":"2025-06-04T19:40:28.442058+01:00"}}
